hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPOfferService.java

```diff
@@ -291,9 +291,8 @@ public String toString() {
   void reportBadBlocks(ExtendedBlock block,
                        String storageUuid, StorageType storageType) {
     checkBlock(block);
+    ReportBadBlockAction rbbAction = new ReportBadBlockAction(block, storageUuid, storageType);
```

Comment:
Is it necessary to optimize this place? it seems worthless.

Reply:
I am not worried about that, because `BPOfferService` is isolated for different namespaces. But I am wonder what improvement here, save the cost to create object and heap footprint?

Reply:
Thank you very much for your explanation! Personally, I believe this modification seems a bit forced and unnecessary. 
Let's wait for @2005hithlj  to give a detailed explanation.

Reply:
@slfan1989  This improvement may not have much effect, but it does reduce the creation of redundant and useless temporary objects. At the same time, from the perspective of code style, the implementation here is not elegant.